.github/workflows/build-linux.yml

```diff
@@ -0,0 +1,214 @@
+name: Build Linux
+
+on:
+  workflow_call:
+    inputs:
+      binary_list:
+        required: true
+        type: string
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        arch: [aarch64, x86_64]
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        submodules: recursive
+
+    - name: Extract version from tag
+      id: get_version
+      run: |
+        VERSION=${GITHUB_REF#refs/tags/}
+        VERSION=${VERSION#v}
+        echo "VERSION=$VERSION" >> $GITHUB_OUTPUT
+        echo "Version extracted: $VERSION"
+
+    - name: Set up Rust
+      uses: dtolnay/rust-toolchain@stable
+      with:
+        toolchain: stable
+        targets: ${{ matrix.arch }}-unknown-linux-gnu
+
+    - name: Install cross-compilation tools
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y gcc-aarch64-linux-gnu g++-aarch64-linux-gnu
+
+    - name: Install dependencies
+      run: |
+        sudo apt-get install -y debhelper zip imagemagick
+        if [ "${{ matrix.arch }}" = "x86_64" ]; then
+          sudo apt-get install -y rpm
+        fi
+
+    - name: Build
+      run: |
+        cargo build --release --target ${{ matrix.arch }}-unknown-linux-gnu
+
+    - name: Create Debian package for GUI
+      run: |
+        mkdir -p debian-gui/DEBIAN
+        mkdir -p debian-gui/usr/bin
+        mkdir -p debian-gui/usr/share/applications
+        mkdir -p debian-gui/usr/share/icons/hicolor/512x512/apps
+        mkdir -p debian-gui/usr/share/icons/hicolor/256x256/apps
+        mkdir -p debian-gui/usr/share/icons/hicolor/128x128/apps
+        mkdir -p debian-gui/usr/share/icons/hicolor/64x64/apps
+        cp target/${{ matrix.arch }}-unknown-linux-gnu/release/node-gui debian-gui/usr/bin/
+        
+        # Copy and convert icon files
+        cp build-tools/assets/node-gui-icon_512.png debian-gui/usr/share/icons/hicolor/512x512/apps/mintlayer-node-gui.png
+        convert build-tools/assets/node-gui-icon_512.png -resize 256x256 debian-gui/usr/share/icons/hicolor/256x256/apps/mintlayer-node-gui.png
+        convert build-tools/assets/node-gui-icon_512.png -resize 128x128 debian-gui/usr/share/icons/hicolor/128x128/apps/mintlayer-node-gui.png
+        convert build-tools/assets/node-gui-icon_512.png -resize 64x64 debian-gui/usr/share/icons/hicolor/64x64/apps/mintlayer-node-gui.png
+        
+        # Create .desktop file
+        cat << EOF > debian-gui/usr/share/applications/mintlayer-node-gui.desktop
+        [Desktop Entry]
+        Name=Mintlayer Node GUI
+        Exec=/usr/bin/node-gui
```

Comment:
I agree with you; binaries should have a prefix to avoid conflicts when installed system-wide. However, implementing this change would break any existing scripts. It's a delicate matter, and I'm not sure we can make this change easily now.

Reply:
> I agree with you; binaries should have a prefix to avoid conflicts when installed system-wide. However, implementing this change would break any existing scripts. It's a delicate matter, and I'm not sure we can make this change easily now.

I wonder whether we should publish the deb/rpm at all in such a case. At least for the command-line tools. Because not only those binaries have generic names, their `--help` output doesn't even mention they are related to Mintlayer. So users may not have any idea what has just been installed into their system.